auth/utils/read-external-sources.ts

```diff
@@ -97,8 +96,31 @@ function filterPropertiesByServiceName(envObj: any, serviceName: string): any {
  * Pulls credentials from VCAP_SERVICES env property that IBM Cloud sets
  *
  */
-function getCredentialsFromCloud(serviceName: string): any {
-  const credentials = vcapServices.getCredentials(serviceName);
+function getVCAPCredentialsForService(name) {
+  if (process.env.VCAP_SERVICES) {
+    const services = JSON.parse(process.env.VCAP_SERVICES);
+    for (const serviceName of Object.keys(services)) {
+      for (const instance of services[serviceName]) {
+        if (instance['name'] === name) {
+          return instance.credentials || {};
+        }
+      }
+    }
+    for (const serviceName in services) {
+      if (serviceName === name) {
+        if (services[serviceName].length > 0) {
+          return services[serviceName][0].credentials || {};
+        } else {
+          return {}
```

Comment:
Speaking of the logger, I think it's a good idea to add debug or verbose logging statements before an empty object is returned saying something like "no data read from vcap services"